File: sdk/kotlin/src/main/kotlin/com/seocheon/sdk/infrastructure/signing/SigningService.kt
package com.seocheon.sdk.infrastructure.signing

/**
 * Interface for transaction signing.
 */
interface SigningService {
    /** Signs the given transaction bytes and returns the signature. */
    suspend fun sign(txBytes: ByteArray): ByteArray

    /** Returns the signer's bech32 address. */
    fun getAddress(): String

    /** Returns the signer's compressed public key bytes. */
    fun getPubKey(): ByteArray
}

File: sdk/swift/Sources/SeocheonSDK/Infrastructure/Signing/SigningService.swift
import Foundation

/// Protocol for transaction signing services.
public protocol SigningService: Sendable {
    /// Signs the given transaction bytes and returns the signature.
    func sign(_ txBytes: Data) async throws -> Data
    /// Returns the signer's bech32 address.
    func getAddress() -> String
    /// Returns the signer's compressed public key bytes.
    func getPubKey() -> Data
}
